### PR TITLE
feat: Add `Stack` and `positioned` widget support

### LINKED ISSUE #113: feat: Add flutter `Stack` support in `Mirai`
**Description**
According to official documentation, Stack is a widget that positions its children relative to the edges of its box.

This class is useful if you want to overlap several children in a simple way, for example having some text and an image, overlaid with a gradient and a button attached to the bottom.

**Additional Context**

For more details please visit the official documentation [ here ](https://api.flutter.dev/flutter/widgets/Stack-class.html)

### PR DESCRIPTION
**Pull request by loud-at-heart**
## Description

According to official documentation, Stack is a widget that positions its children relative to the edges of its box.

This class is useful if you want to overlap several children in a simple way, for example having some text and an image, overlaid with a gradient and a button attached to the bottom.

In this PR, Mirai Stack has been added to parse Stack Parameter

## Related Issues

closes #113 

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code refactor
- [ ] Build configuration change
- [ ] Documentation
- [ ] Chore

### CODE REVIEW

**Comment by loud-at-heart** [comment]
@i-asimkhan Please do have a check on this

File: packages/mirai/lib/src/parsers/mirai_stack/mirai_stack_parser.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import 'package:mirai/src/utils/widget_type.dart';
+
+class MiraiStackParser extends MiraiParser<MiraiStack> {
+  const MiraiStackParser();
+
+  @override
+  MiraiStack getModel(Map<String, dynamic> json) => MiraiStack.fromJson(json);
+
+  @override
+  String get type => WidgetType.column.name;
```
Comment (i-asimkhan) [inline packages/mirai/lib/src/parsers/mirai_stack/mirai_stack_parser.dart:13]: We need to add a new widget type called stack and use it here.

File: packages/mirai/lib/src/parsers/mirai_gradient/mirai_gradient.dart [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import 'dart:math' as math;
+
 import 'package:flutter/material.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
 import 'package:mirai/src/parsers/parsers.dart';
 import 'package:mirai/src/utils/color_utils.dart';
 
 part 'mirai_gradient.freezed.dart';
+
```
Comment (i-asimkhan) [inline packages/mirai/lib/src/parsers/mirai_gradient/mirai_gradient.dart:9]: Can we please remove this extra line?

**Review by i-asimkhan — COMMENTED** [review]
Thanks @loud-at-heart It is awesome feature, please can you add an example for this widget just like this https://github.com/Securrency-OSS/mirai/blob/main/examples/mirai_gallery/assets/json/container_example.json

File: packages/mirai/lib/src/parsers/mirai_gradient/mirai_gradient.dart (hunk 2 above)
Comment (loud-at-heart) [reply packages/mirai/lib/src/parsers/mirai_gradient/mirai_gradient.dart]: The extra line has been removed

File: packages/mirai/lib/src/parsers/mirai_stack/mirai_stack_parser.dart (hunk 1 above)
Comment (loud-at-heart) [reply packages/mirai/lib/src/parsers/mirai_stack/mirai_stack_parser.dart]: New widgetType as stack has been added

**Comment by loud-at-heart** [comment]
@i-asimkhan since stack and positioned go hand in hand so I have added support for positioned also. The examples have been added as per asked

File: packages/mirai/lib/src/parsers/mirai_stack/mirai_stack_parser.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  @override
+  Widget parse(BuildContext context, MiraiStack model) {
+    return Stack(
+      alignment: model.alignment.valueByDirection,
+      clipBehavior: model.clipBehavior,
+      fit: model.fit,
+      textDirection: model.textDirection,
+      children: model.children
+          .map(
+            (value) => Mirai.fromJson(value, context) ?? const SizedBox(),
```
Comment (divyanshub024) [inline packages/mirai/lib/src/parsers/mirai_stack/mirai_stack_parser.dart:24]: Can we remove the `comma(,)` from the end for better code formatting and readability?

**Review by divyanshub024 — CHANGES_REQUESTED** [review]
This is pretty great work 💯 
Please check my suggestions.

File: packages/mirai/lib/src/parsers/mirai_stack/mirai_stack_parser.dart (hunk 3 above)
Comment (loud-at-heart) [reply packages/mirai/lib/src/parsers/mirai_stack/mirai_stack_parser.dart]: Changes have been made as per asked

File: packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned.dart [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:mirai/src/parsers/mirai_rect/mirai_rect.dart';
+
+export 'package:mirai/src/parsers/mirai_positioned/mirai_positioned_parser.dart';
+
+part 'mirai_positioned.freezed.dart';
+part 'mirai_positioned.g.dart';
+
+/*
+* TODO :: Add support for fromRelativeRect
```
[the remark this answers is not in the document]
Comment (loud-at-heart) [reply packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned.dart:11]: If this method is necessary, then I will code for this?

File: packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned_parser.dart [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        );
+      default:
+        return Positioned(
+          left: model.left,
+          top: model.top,
+          right: model.right,
+          bottom: model.bottom,
+          height: model.height,
+          width: model.width,
+          child: Mirai.fromJson(model.child, context) ?? const Placeholder(),
```
[the remark this answers is not in the document]
Comment (loud-at-heart) [reply packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned_parser.dart]: Changes have been made as per asked

File: packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned_parser.dart [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+          left: model.left,
+          top: model.top,
+          right: model.right,
+          bottom: model.bottom,
+          child: Mirai.fromJson(model.child, context) ?? const Placeholder(),
+        );
+      case MiraiPositionedType.fromRect:
+        return Positioned.fromRect(
+          rect: model.rect?.parse ?? Rect.zero,
+          child: Mirai.fromJson(model.child, context) ?? const Placeholder(),
```
[the remark this answers is not in the document]
Comment (loud-at-heart) [reply packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned_parser.dart]: Changes have been made as per asked

File: packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned_parser.dart [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+          width: model.width,
+          child: Mirai.fromJson(model.child, context) ?? const Placeholder(),
+        );
+      case MiraiPositionedType.fill:
+        return Positioned.fill(
+          left: model.left,
+          top: model.top,
+          right: model.right,
+          bottom: model.bottom,
+          child: Mirai.fromJson(model.child, context) ?? const Placeholder(),
```
[the remark this answers is not in the document]
Comment (loud-at-heart) [reply packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned_parser.dart]: Changes have been made as per asked

File: packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned_parser.dart [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  Widget parse(BuildContext context, MiraiPositioned model) {
+    switch (model.positionedType) {
+      case MiraiPositionedType.directional:
+        return Positioned.directional(
+          textDirection: model.textDirection,
+          top: model.top,
+          bottom: model.bottom,
+          height: model.height,
+          width: model.width,
+          child: Mirai.fromJson(model.child, context) ?? const Placeholder(),
```
[the remark this answers is not in the document]
Comment (loud-at-heart) [reply packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned_parser.dart]: Changes have been made as per asked

File: packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned_parser.dart [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  @override
+  String get type => WidgetType.positioned.name;
+
+  @override
+  Widget parse(BuildContext context, MiraiPositioned model) {
+    switch (model.positionedType) {
+      case MiraiPositionedType.directional:
+        return Positioned.directional(
+          textDirection: model.textDirection,
+          top: model.top,
```
[the remark this answers is not in the document]
Comment (loud-at-heart) [reply packages/mirai/lib/src/parsers/mirai_positioned/mirai_positioned_parser.dart]: The above params were added

File: packages/mirai/lib/src/parsers/mirai_alignment/mirai_alignment.dart [hunk 10]
```diff
@@ -9,9 +9,15 @@ enum MiraiAlignment {
...
   bottomLeft,
   bottomCenter,
-  bottomRight;
+  bottomRight,
+  topStart,
+  topEnd,
+  centerStart,
+  centerEnd,
+  bottomStart,
+  bottomEnd;
```
[the remark this answers is not in the document]
Comment (loud-at-heart) [reply packages/mirai/lib/src/parsers/mirai_alignment/mirai_alignment.dart]: Changes have been made as per asked. A new file `MiraiAlignmentDirectional` has been implemented.

File: examples/mirai_gallery/assets/json/home_screen.json [hunk 11]
```diff
@@ -201,6 +201,41 @@
...
+                    "subtitle": {
+                        "type": "text",
+                        "data": "This is a mirai stack widget.",
+                        "align": "center",
+                        "style": {
+                            "fontSize": 12
+                        }
+                    },
+                    "isThreeLine": true,
+                    "style": "list",
```
[the remark this answers is not in the document]
Comment (loud-at-heart) [reply examples/mirai_gallery/assets/json/home_screen.json]: Changes have been made as per asked

File: examples/mirai_gallery/assets/json/home_screen.json [hunk 12]
```diff
@@ -201,6 +201,41 @@
...
+                    "type": "listTile",
+                    "leading": {
+                        "type": "icon",
+                        "iconType": "material",
+                        "icon": "layers"
+                    },
+                    "title": {
+                        "type": "text",
+                        "data": "Mirai Stack",
+                        "align": "center",
```
[the remark this answers is not in the document]
Comment (loud-at-heart) [reply examples/mirai_gallery/assets/json/home_screen.json]: Changes have been made as per asked

**Review by loud-at-heart — COMMENTED** [review]
The changes have been made as per asked

**Review by divyanshub024 — APPROVED** [review]
Looks great 🚀

**Pull request merged**